roachtest: rust-postgres failed

**Note:** This build has runtime assertions enabled. If the same failure was hit in a run without assertions enabled, there should be a similar failure without this message. If there isn&#39;t one, then this failure is likely due to an assertion violation or (assertion) timeout.

roachtest.rust-postgres [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17126933?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/17126933?buildTab=artifacts#/rust-postgres) on release-24.1.6-rc @ [63dd3482501dbc49e9e315c02c17c28b117b651e](https://github.com/cockroachdb/cockroach/commits/63dd3482501dbc49e9e315c02c17c28b117b651e):


```
(orm_helpers.go:229).summarizeFailed: 
Tests run on Cockroach v24.1.6-dev-63dd3482501dbc49e9e315c02c17c28b117b651e
Tests run against rust-postgres 
143 Total Tests Run
75 tests passed
68 tests failed
0 tests skipped
1 test ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: test.copy_out (unexpected)
For a full summary look at the rust-postgres artifacts 
An updated blocklist (rustPostgresBlockList) is available in the artifacts' rust-postgres log
test artifacts and logs in: /artifacts/rust-postgres/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=16</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_runtimeAssertionsBuild=true</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-17126933/rust-postgres/1727848939791/1727849126055)

</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*rust-postgres.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42660

--- a/pkg/sql/copy/BUILD.bazel
+++ b/pkg/sql/copy/BUILD.bazel
@@ -13,6 +13,7 @@ go_test(
     shard_count = 4,
     deps = [
         "//pkg/base",
+        "//pkg/ccl",
         "//pkg/cli/clisqlclient",
         "//pkg/kv",
         "//pkg/kv/kvclient/kvcoord",

--- a/pkg/sql/copy/copy_test.go
+++ b/pkg/sql/copy/copy_test.go
@@ -22,6 +22,7 @@ import (
 
 	"github.com/cockroachdb/apd/v3"
 	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/ccl"
 	"github.com/cockroachdb/cockroach/pkg/cli/clisqlclient"
 	"github.com/cockroachdb/cockroach/pkg/kv"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
@@ -84,6 +85,7 @@ const csvData = `%d|155190|7706|1|17|21168.23|0.04|0.02|N|O|1996-03-13|1996-02-1
 
 func TestDataDriven(t *testing.T) {
 	defer leaktest.AfterTest(t)()
+	defer ccl.TestingEnableEnterprise()() // allow usage of READ COMMITTED
 	ctx := context.Background()
 
 	doTest := func(t *testing.T, d *datadriven.TestData, conn clisqlclient.Conn) string {

--- a/pkg/sql/copy/testdata/copy_to_read_committed
+++ b/pkg/sql/copy/testdata/copy_to_read_committed
@@ -1,0 +1,27 @@
+# Verify that the default transaction isolation level is applied to the
+# internal queries issued by the COPY command.
+exec-ddl
+SET default_transaction_isolation = 'read committed'
+----
+
+copy-to
+COPY (SELECT current_setting('transaction_isolation')) TO STDOUT
+----
+read committed
+
+copy-to
+COPY (SELECT 1, 2, 3) TO STDOUT
+----
+1	2	3
+
+copy-to
+COPY (
+  SELECT repeat('a' || i::STRING, 5), repeat('b' || i::STRING, 5)
+  FROM ROWS FROM (generate_series(1, 5)) AS i
+) TO STDOUT;
+----
+a1a1a1a1a1	b1b1b1b1b1
+a2a2a2a2a2	b2b2b2b2b2
+a3a3a3a3a3	b3b3b3b3b3
+a4a4a4a4a4	b4b4b4b4b4
+a5a5a5a5a5	b5b5b5b5b5

--- a/pkg/sql/copy_to.go
+++ b/pkg/sql/copy_to.go
@@ -181,10 +181,10 @@ func runCopyTo(
 		}).String()
 	}
 
-	it, err := p.execCfg.InternalDB.NewInternalExecutor(p.SessionData()).QueryIteratorEx(
+	it, err := p.InternalSQLTxn().QueryIteratorEx(
 		ctx,
 		cmd.Stmt.String(),
-		txn,
+		p.Txn(),
 		sessiondata.NoSessionDataOverride,
 		q,
 	)

--- a/pkg/sql/isql/isql_db.go
+++ b/pkg/sql/isql/isql_db.go
@@ -262,9 +262,6 @@ type Rows interface {
 	// Types returns the types of the columns returned by this iterator. The
 	// returned array is guaranteed to correspond 1:1 with the tree.Datums rows
 	// returned by Cur().
-	//
-	// WARNING: this method is safe to call anytime *after* the first call to
-	// Next() (including after Close() was called).
 	Types() colinfo.ResultColumns
 
 	// HasResults returns true if there are results to the query, false otherwise.

